pyQuARC/code/custom_validator.py

```diff
@@ -179,11 +179,11 @@ def uniqueness_check(list_of_objects, key):
         seen, duplicates = set(), set()
         if isinstance(list_of_objects, list):
             for url_obj in list_of_objects:
-                if description := url_obj.get(key) in seen:
+                description = url_obj.get(key)
+                if description in seen:
```

Comment:
Why the change?

Reply:
Using the following valid/invalid values from checks_umm-g_master_test_file.json:
```
    "url_description_uniqueness_check": {
        "valid": [
            [
                {
                    "Description": "Example description 1",
                    "Subtype": "GENERAL DOCUMENTATION",
                    "Type": "VIEW RELATED INFORMATION",
                    "URL": "https://www.nasa.gov/",
                    "URLContentType": "PublicationURL"
                },
                {
                    "Description": "Example description 2",
                    "Subtype": "PUBLICATIONS",
                    "Type": "VIEW RELATED INFORMATION",
                    "URL": "https://www.nasa.gov/",
                    "URLContentType": "PublicationURL"
                }
            ],
            "Description"],
        "invalid": [
            [
                {
                    "Description": "Example description 1",
                    "Subtype": "GENERAL DOCUMENTATION",
                    "Type": "VIEW RELATED INFORMATION",
                    "URL": "https://www.nasa.gov/",
                    "URLContentType": "PublicationURL"
                },
                {
                    "Description": "Example description 1",
                    "Subtype": "PUBLICATIONS",
                    "Type": "VIEW RELATED INFORMATION",
                    "URL": "https://www.nasa.gov/",
                    "URLContentType": "PublicationURL"
                }
            ],
            "Description"]
    },

```
Example test output for umm-g with
`                if description := url_obj.get(key) in seen:`
```
rule_name: URL Description Uniqueness Check
check_id: uniqueness_check
validator function: CustomValidator.uniqueness_check
test output:
with valid test input: {'valid': True, 'value': ''}
with invalid test input: {'valid': True, 'value': ''}
```
Example test output for umm-g with
```
                description = url_obj.get(key)
                if description in seen:

```
```
rule_name: URL Description Uniqueness Check
check_id: uniqueness_check
validator function: CustomValidator.uniqueness_check
test output:
with valid test input: {'valid': True, 'value': ''}
with invalid test input: {'valid': False, 'value': 'Example description 1'}

```

Reply:
it might just be because of missing parenthesis. i.e. `if (description := url_obj.get(key)) in seen:`, see if that works?